src/dmi.rs

```diff
@@ -28,3 +31,25 @@ fn write_png(path: &str, info: OutputInfo, image: Vec<u8>) -> Result<()> {
     let mut writer = encoder.write_header()?;
     Ok(writer.write_image_data(&image)?)
 }
+
+fn create_png(path: &str, width: &str, height: &str, data: &str) -> Result<()> {  
+    let width = u32::from_str_radix(width,10)?;
+    let height = u32::from_str_radix(height,10)?;
+    
+    let mut result : Vec<u8> = Vec::new();
+    for chunk in data.chars().collect::<Vec<char>>().chunks(7) {
+        let single: String = chunk.into_iter().collect();
```

Comment:
I feel like you could do this without as much `collect()`

Reply:
Alright, changed it to SO suggestion, i feel getting rid of inner collect would make this bit unreadable.